### PR TITLE
[Onboarding] Adds "Agree to Terms" to sign up flow for GDPR compliance

### PR DESCRIPTION
**Pull request by l2succes**
TODO
- [ ] Need to fix *Terms of Use* and *Privacy Policy* links
- [ ] Swap check/unchecked icons

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">No CHANGELOG changes made</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is classed as Work in Progress</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_travis">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: Artsy/App_Resources/Artsy-Info.plist [hunk 1]
```diff
@@ -42,8 +42,11 @@
 	<key>CFBundleURLTypes</key>
 	<array>
 		<dict>
+			<key>CFBundleTypeRole</key>
+			<string>Editor</string>
 			<key>CFBundleURLSchemes</key>
 			<array>
+				<string>fb-</string>
```
Comment (ashfurrow) [inline Artsy/App_Resources/Artsy-Info.plist]: Would this register our app for `fb://` URLs?

File: Artsy/View_Controllers/Login_and_Onboarding/AROnboardingViewController.m [hunk 2]
```diff
@@ -43,9 +43,11 @@ @interface AROnboardingViewController () <UINavigationControllerDelegate>
 @property (nonatomic, strong, readwrite) NSLayoutConstraint *iPhoneXStatusBarHeightConstraint;
 @property (nonatomic, strong, readwrite) UIView *progressBar;
 @property (nonatomic, strong, readwrite) UIView *progressBackgroundBar;
+@property (nonatomic, strong, readwrite) NSString *name;
 @property (nonatomic, strong, readwrite) NSString *email;
 @property (nonatomic, strong, readwrite) NSString *password;
 @property (nonatomic, nonnull, strong, readwrite) UITextField *tempTextField;
+@property (nonatomic) BOOL shouldPresentFacebook;
```
Comment (ashfurrow) [inline Artsy/View_Controllers/Login_and_Onboarding/AROnboardingViewController.m]: I know that properties are `assign` by default, but the rest of Eigen explicitly specifies that. I'd change this just to be consistent.

File: Artsy/App_Resources/Artsy-Info.plist (hunk 1 above)
Comment (l2succes) [reply Artsy/App_Resources/Artsy-Info.plist]: We get this error from `FBSDKLoginManager` if it's not included.

```
2018-05-09 16:56:09.952957-0400 Artsy[19846:158232812] *** Terminating app due to uncaught exception 'InvalidOperationException', reason: 'fb- is not registered as a URL scheme. Please add it in your Info.plist'
*** First throw call stack:
(
	0   CoreFoundation                      0x000000011377f1ab __exceptionPreprocess + 171
	1   libobjc.A.dylib                     0x00000001121ebf41 objc_exception_throw + 48
	2   Artsy                               0x000000010a890859 +[FBSDKInternalUtility validateURLSchemes] + 393
	3   Artsy                               0x000000010a8be4ad -[FBSDKLoginManager logInParametersWithPermissions:serverConfiguration:] + 109
	4   Artsy                               0x000000010a8bed67 -[FBSDKLoginManager logInWithBehavior:] + 103
	5   Artsy                               0x000000010a8becbc -[FBSDKLoginManager logInWithPermissions:handler:] + 380
	6   Artsy                               0x000000010a8bcba2 -[FBSDKLoginManager logInWithReadPermissions:fromViewController:handler:] + 386
```

File: Artsy/View_Controllers/Login_and_Onboarding/AROnboardingViewController.m (hunk 2 above)
Comment (l2succes) [reply Artsy/View_Controllers/Login_and_Onboarding/AROnboardingViewController.m]: Sounds good, i'll do that

**Comment by l2succes** [comment]
🙏🏾 Thanks @orta

File: Artsy/App_Resources/Artsy-Info.plist (hunk 1 above)
Comment (ashfurrow) [reply Artsy/App_Resources/Artsy-Info.plist]: Facebook's SDK never fails to surprise!

**Pull request merged**